### PR TITLE
Add support for Amazon lambda tests

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: extensions/amazon-lambda/runtime/src/main/java/io/quarkus/amazon/lambda/runtime/AmazonLambdaTemplate.java
Comment (gsmet) [inline extensions/amazon-lambda/runtime/src/main/java/io/quarkus/amazon/lambda/runtime/AmazonLambdaTemplate.java]: We really should use a logger here and a few lines below (`ex.printStackTrace();`)

**Review by gsmet — CHANGES_REQUESTED** [review]
I'll let Justin review it but I added one comment.

**Comment by mkouba** [comment]
Looks good. @gsmet ok to merge?

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
@mkouba done, thanks for the reminder.

I'm a bit worried about us having hardcoded URLs such as `/2018-06-01/runtime/init/error` but I suppose it can be considered API from the Lambda POV.

In any case, they were already present so it shouldn't block this PR.

**Comment by mkouba** [comment]
> I'm a bit worried about us having hardcoded URLs

Yes, I'd like to have it configurable or at least use shared constants.

**Pull request merged**
